server/item/armour_trim.go

```diff
@@ -0,0 +1,58 @@
+package item
+
+import "github.com/df-mc/dragonfly/server/world"
+
+type ArmourTrim struct {
+	Template ArmourSmithingTemplate
+	Material ArmourTrimMaterial
+}
+
+type ArmourTrimMaterial interface {
+	// TrimMaterial returns the material name used for reading and writing trim data.
+	TrimMaterial() string
+	// MaterialColour returns the color code used for internal text formatting. Use text.Colourf for proper formatting.
```

Comment:
Change "color" to "colour", can probably remove the `text.Colourf` reference now we return the constant